x/masterchef/keeper/query_chain_tvl.go

```diff
@@ -0,0 +1,62 @@
+package keeper
+
+import (
+	"context"
+	"cosmossdk.io/math"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	ptypes "github.com/elys-network/elys/x/parameter/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/elys-network/elys/x/masterchef/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k Keeper) ChainTVL(goCtx context.Context, req *types.QueryChainTVLRequest) (*types.QueryChainTVLResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	allPools := k.amm.GetAllPool(ctx)
+	poolsTVL := math.LegacyZeroDec()
+	totalTVL := math.ZeroInt()
+
+	for _, pool := range allPools {
+		tvl, err := pool.TVL(ctx, k.oracleKeeper, k.accountedPoolKeeper)
+		if err != nil {
+			return nil, err
+		}
+		poolsTVL = poolsTVL.Add(tvl)
+	}
+	totalTVL = totalTVL.Add(poolsTVL.TruncateInt())
+
+	baseCurrencyEntry, found := k.assetProfileKeeper.GetEntry(ctx, ptypes.BaseCurrency)
+	if !found {
+		return nil, status.Error(codes.NotFound, "asset profile not found")
+	}
+
+	stableStakeTVL := k.stableKeeper.TVL(ctx, k.oracleKeeper, baseCurrencyEntry.Denom)
+	totalTVL = totalTVL.Add(stableStakeTVL.TruncateInt())
```

Comment:
this seems to be repeated code, can you reuse the function above for this ?

Reply:
two different queries, one is chain tvl and other pool tvl query as required by defi lama